packages/frontend/src/components/chain-select/chain-select.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { ChainIcon } from "../chain-icon";
+import { ENABLED_CHAINS, SUPPORTED_CHAINS } from "../../constants";
+import type { ChainId } from "@carrot-kpi/sdk";
+import CaretDown from "../../icons/caret-down";
+import { useAccount, useNetwork } from "wagmi";
+import Error from "../../icons/error";
+import { NetworksPopover } from "./networks-popover";
+import { Typography } from "@carrot-kpi/ui";
+import { useTranslation } from "react-i18next";
+
+export interface ChainSelectProps {
+    compact?: boolean;
+}
+
+export const ChainSelect = ({ compact = true }: ChainSelectProps) => {
+    const { t } = useTranslation();
+    const { chain } = useNetwork();
+    const { address, connector: activeConnector } = useAccount();
+
+    const [networksPopoverAnchor, setNetworksPopoverAnchor] =
+        useState<HTMLDivElement | null>(null);
+    const [networksPopoverOpen, setNetworksPopoverOpen] = useState(false);
+
+    const networksPopoverRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        const handleMouseDown = (event: MouseEvent) => {
+            if (
+                networksPopoverRef.current &&
+                !networksPopoverRef.current.contains(event.target as Node)
+            )
+                setNetworksPopoverOpen(false);
+        };
+        document.addEventListener("mousedown", handleMouseDown);
+        return () => {
+            window.removeEventListener("mousedown", handleMouseDown);
```

Comment:
I noticed we add the event listener to `document` and then we remove it from `window`. Not 100% sure, but I guess this could result in the listener still being active when the component unmounts.

Reply:
Nice catch, as suggested below I've replaced this with `useClickAway`, fixed in 4b0532268fb3507753b83ec7a80e92c388c81520.